src/components/modal/ModalSelect.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Icon, Text, View } from 'native-base';
+import React from 'react';
+import Modal from 'react-native-modal';
+import computeFormStyleSheet from '../../FormStyles';
+import BaseProps from '../../types/BaseProps';
+import Utils from '../../utils/Utils';
+import { ItemSelectionMode } from '../list/ItemsList';
+import computeStyleSheet from './ModalStyles';
+import I18n from 'i18n-js';
+import SelectableList from '../../screens/base-screen/SelectableList';
+
+export interface Props<T> extends BaseProps {
+  defaultItem: T;
+  buildItemName: (item: T) => string;
+  selectionMode: ItemSelectionMode;
+  onItemsSelected: (selectedItems: T[]) => void;
+}
+interface State<T> {
+  isVisible: boolean;
+  selectedItems: T[];
+  canValidateMultiSelection: boolean;
+}
+
+export default class ModalSelect<T> extends React.Component<Props<T>, State<T>> {
+  public state: State<T>;
+  public props: Props<T>;
+  private itemsListRef = React.createRef<SelectableList<T>>();
+  public constructor(props: Props<T>) {
+    super(props);
+    this.state = {
+      isVisible: false,
+      selectedItems: [],
+      canValidateMultiSelection: false
+    };
+  }
+
+  public render() {
+    const style = computeStyleSheet();
+    const formStyle = computeFormStyleSheet();
+    const { buildItemName, selectionMode, defaultItem } = this.props;
+    const { selectedItems, isVisible, canValidateMultiSelection } = this.state;
+    const itemsList = React.Children.only(this.props.children);
+    const otherSelectedItemsCount = Math.max(selectedItems?.length - 1, 0);
+    const initializedSelectedItems = Utils.isEmptyArray(selectedItems) ? [defaultItem] : selectedItems;
```

Comment:
You should let the default to an empty array, otherwise you will always enforce a default selection even if the user does not want one.

Reply:
We always want a default user (the current logged in user as discussed with Serge) because we need at least one user to start a transaction for

Reply:
The component code is targeting a generic usage but its API is tied to a specific use case. That's not a clean API, the API must handle all possible use case: with or without default value selection.

Reply:
I made it optional, so when we don't pass it we don't have a default selection